x-pack/plugins/apm/server/routes/source_maps/create_source_map_index.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+import {
+  createOrUpdateIndex,
+  Mappings,
+} from '@kbn/observability-plugin/server';
+import { getApmIndicesConfig } from '../settings/apm_indices/get_apm_indices';
+import { APMConfig } from '../..';
+
+export const createSourceMapIndex = async ({
```

Comment:
Why not save it on a saved object instead?

Reply:
It crossed my mind, and it's a good idea always to consider instead of "self managing". In this case APM Server needs to access the indices directly and won't access them through Kibana. Since Saved Objects are stored deep inside `.kibana` accessing them directly is both awkward and recommended against.

We have prior art for doing this (agent configs and custom links) which both have turned out well, without too many issues coming from the field (aka sdh's). So I trust the concept enough to go ahead with this.

As a rule of thumb I'd use saved objects for anything that needs to be space aware (like apm indices and service groups) and where we will query it through Kibana. Whenever we want to advanced aggregations and custom queries we should consider using a "self managed" index.